src/utils/outputs/style-dictionary/colors.ts

```diff
@@ -0,0 +1,149 @@
+import { DefaultColorType, HeisslufType, SpeakingName } from "../../data.ts";
+import { getPalette } from "../index.ts";
+import { setObjectByPath } from "./index.ts";
+
+export const getSDColorPalette = (
+  allColors: Record<string, DefaultColorType>,
+  luminanceSteps: number[],
+): any => {
+  const palette: Record<string, HeisslufType[]> = getPalette(
+    allColors,
+    luminanceSteps,
+  );
+  const colors: any = {};
+
+  Object.entries(allColors).forEach(([unformattedName, color]) => {
+    const name = unformattedName.toLowerCase();
+
+    const colorValues: any = {};
+
+    const hslType: HeisslufType[] = palette[unformattedName];
+    hslType.forEach((hsl) => {
+      colorValues[`${hsl.index ?? hsl.name}`] = { value: hsl.hex };
+    });
+
+    colorValues.origin = {
+      base: {
+        value: color.origin,
+      },
+      light: {
+        default: {
+          value: color.originLight,
+        },
+        hovered: {
+          value: color.originLightHovered,
+        },
+        pressed: {
+          value: color.originLightPressed,
+        },
+      },
+      dark: {
+        default: {
+          value: color.originDark,
+        },
+        hovered: {
+          value: color.originDarkHovered,
+        },
+        pressed: {
+          value: color.originDarkPressed,
+        },
+      },
+    };
+
+    colorValues.onOrigin = {
+      light: {
+        default: {
+          value: color.onOriginLight,
+        },
+        hovered: {
+          value: color.onOriginLightHovered,
+        },
+        pressed: {
+          value: color.onOriginLightPressed,
+        },
+      },
+      dark: {
+        default: {
+          value: color.onOriginDark,
+        },
+        hovered: {
+          value: color.onOriginDarkHovered,
+        },
+        pressed: {
+          value: color.onOriginDarkPressed,
+        },
+      },
+    };
+
+    colors[name] = colorValues;
+  });
+
+  return { colors };
+};
+
+export const getSDSpeakingColors = (
```

Comment:
**question**: Is `SpeakingColors` an existing term in this application?

I did not find any naming in the [Design Tokens W3C Spec](https://design-tokens.github.io/community-group/format/), but the [Figma Docs](https://help.figma.com/hc/en-us/articles/18490793776023-Update-1-Tokens-variables-and-styles#semantic-tokens) (and I 😄) use the name _semantic tokens_.

Reply:
We use the term "semantic" for the colors `successful`, `critical`, `informational` and `warning`.
To avoid misunderstandings we decided to use "speaking-colors" for those tokens.

I can discuss this with our design colleagues, why they use the terms like this if you like

Reply:
This would be interesting to know the thoughts behind the naming. "speaking color" seems to me like a failed translation from german to english.

I use the following token types:

- **core tokens**: hardcoded values  e.g. `colors.red.500 = #f00`
- **semantic tokens**: tokens, which can alter their value depending of the surrounding context e.g. (color mode, parent dom node, user preferences)
- **component tokens**: tokens, which can alter their value based on the behaviour of a component e.g. checked checkbox, collapsed accordion etc.

Reply:
I asked a colleague, she said it was "grown historically" ^.^

Like I said they reserved the term "semantic" for another context. But we want to change the "speaking color" to "usage colors" (tbd. with the rest of the team).

Would it be a big problem on your side if the filename is `usage-colors.json`?

Reply:
Thanks for the info!

> Would it be a big problem on your side if the filename is `usage-colors.json`?

No, filenames do not matter to Style Dictionary.